Makefile

```
@@ -617,9 +617,9 @@ release-compress: | $(DIST_DIRS)
 	cd $(DIST)/release/; for file in `find . -type f -name "*"`; do echo "compressing $${file}" && gxz -k -9 $${file}; done;
 
 .PHONY: release-sources
-release-sources: | $(DIST_DIRS) node_modules
+release-sources: | $(DIST_DIRS) npm-cache
 	echo $(VERSION) > $(STORED_VERSION_FILE)
-	tar --exclude=./$(DIST) --exclude=./.git --exclude=./$(MAKE_EVIDENCE_DIR) --exclude=./node_modules/.cache --exclude=./$(AIR_TMP_DIR) -czf $(DIST)/release/gitea-src-$(VERSION).tar.gz .
+	bsdtar --exclude=^./$(DIST) --exclude=^./.git --exclude=^./$(MAKE_EVIDENCE_DIR) --exclude=node_modules --exclude=^./$(AIR_TMP_DIR) -czf $(DIST)/release/gitea-src-$(VERSION).tar.gz .
```

Is the node_module still included tarball?

---

It looks like it is not.

However, my issue here is with the new hard requirement on bsdtar

There is absolutely no need for regexp matching here - you're simply adding in complication for no reason at all and I worry that you're not even doing it correctly.

```
^./$(MAKE_EVIDENCE_DIR)
```

matches:

```
a/$(MAKE_EVIDENCE_DIR)
b/$(MAKE_EVIDENCE_DIR)
c/$(MAKE_EVIDENCE_DIR)
...
```

and that's without going in to regexp related issues potentially affecting the values of `MAKE_EVIDENCE_DIR`.

Why not just detect if bsdtar is present:

```suggestion
	@hash bsdtar > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
		bsdtar --exclude=^./$(DIST) --exclude=^./.git --exclude=^./$(MAKE_EVIDENCE_DIR) --exclude=node_modules --exclude=^./$(AIR_TMP_DIR) -czf $(DIST)/release/gitea-src-$(VERSION).tar.gz .
	else
		tar --exclude=./$(DIST) --exclude=./.git --exclude=./$(MAKE_EVIDENCE_DIR) --exclude=./node_modules --exclude=./$(AIR_TMP_DIR) -czf $(DIST)/release/gitea-src-$(VERSION).tar.gz .
	fi
```

---

@lunny we need node_modules included for builds not having internet access